src/tilecache.js

```diff
@@ -465,7 +465,7 @@
                     this._destroySelfUnsafe(this._data, this._type);
                 } else if (this._promise) {
                     const oldType = this._type;
-                    this._promise.then(x => this._destroySelfUnsafe(x, oldType));
```

Comment:
Not sure why, but without catch the method in some tests threw error. After adding this part, the error vanished and no console error logs were written...